kernel/kernel/fs/pipe.cpp

```diff
@@ -415,7 +416,7 @@ ssize_t pipe::append_iter(iovec_iter *iter, bool atomic)
         // See if we have space in this pipe buffer
         // TODO: Idea to test: memmove data back if we have offset != 0
         // May compact things a bit.
-        if (PAGE_SIZE - last_buf->len_ <= len)
+        if (PAGE_SIZE - last_buf->len_ <= iter->bytes)
```

Comment:
This function has typos and tiny bugs which should be fixed up and squashed with the previous commit

Reply:
squashed